lib/service.js

```diff
@@ -307,28 +377,33 @@ class Service extends EventEmitter {
    * @return {boolean}
    */
   isTaskHealthy(taskArn) {
-    var task = this.getTask(taskArn);
+    let task = this.getTask(taskArn);
     if (!task) {
       this.emit('warning', `Task ${taskArn} does not exist. Or has not been cached against the service`);
       return false;
     }
 
-    var containerInstance = this.getContainerInstance(task.containerInstanceArn);
+    // Most of this logic is identical for Fargate- and EC2-backed containers,
+    // except for the inner logic around finding the targets.
+    const isHealthy = this.raw.loadBalancers.every((lb) => {
+      const container = task.containers.find((container) => container.name === lb.containerName);
```

Comment:
Do we need to check the value of `container`?

Reply:
I guess the failure condition here would be that we have a task but has no containers defined. Is that even legal?
Previously all of the EC2-based logic had an implicit assumption that the container would be found so it was never checked. I can add an explicit failure condition here though.